file/writable_file_writer.h

```diff
@@ -114,6 +114,17 @@ class WritableFileWriter {
     }
     info.status.PermitUncheckedError();
   }
+
+  void NotifyOnIOError(const IOStatus& io_status, const std::string& operation,
+                       const std::string& file_path, size_t length = 0,
+                       uint64_t offset = 0) {
+    IOErrorInfo io_error_info(io_status, operation, file_path, length, offset);
```

Comment:
Can we only create this inside an "if (!listeners_empty()" ?

Reply:
`ShouldNotifyListeners()` already checks if 'listeners_empty()` or not and then call the `NotifyOnIOError`

Reply:
Right.  I started from here and worked my back to the other code during the review.